### ISSUE TITLE
Newly build inverted index has corruption issue

### ISSUE THREAD

**Issue by yudelevi** [ClickHouse/ClickHouse#58635]
Running on 23.12.2.59 version.
Table def:
``` 
CREATE TABLE scrape.embeddings
(
    `domain` String,
    `embeddings` Array(Float32),
    `text` String,
    INDEX text_idx lower(text) TYPE inverted GRANULARITY 1
)
ENGINE = ReplacingMergeTree()
ORDER BY (domain)
TTL mtime + toIntervalDay(91)
SETTINGS index_granularity = 8192;
```

Error:
```
ch-01 :) select count(*) from scrape.embeddings where (hasToken(lower(text), 'abm')) settings use_skip_indexes=1;

SELECT count(*)
FROM scrape.embeddings
WHERE hasToken(lower(text), 'abm')
SETTINGS use_skip_indexes = 1

Query id: c64aafa7-eb65-43bb-a013-adf74a5fd55b


Elapsed: 0.005 sec. 

Received exception from server (version 23.12.2):
Code: 246. DB::Exception: Received from localhost:9000. DB::Exception: Can't decompress data with codec byte 48 using codec with byte 144. (CORRUPTED_DATA)

ch-01 :) select count(*) from scrape.embeddings where (hasToken(lower(text), 'abm')) settings use_skip_indexes=0;

SELECT count(*)
FROM scrape.embeddings
WHERE hasToken(lower(text), 'abm')
SETTINGS use_skip_indexes = 0

Query id: 417a7d97-2253-457a-b15a-efa7a02485ad

┌─count()─┐
│   18646 │
└─────────┘

1 row in set. Elapsed: 13.045 sec. Processed 69.20 million rows, 268.01 GB (5.31 million rows/s., 20.55 GB/s.)
Peak memory usage: 439.75 MiB.
```
I've just finished building the inverted index, so I doubt it's index corruption.

**Comment by HarryLeeIBM** [comment]
@yudelevi, I tried to reproduce the issue, but it seems working fine when I tried the first SQL(I removed the "TTL" part from the SQL since the 'mtime' is undefined). Was the table created before  23.12.2.59 release? If it was, could you drop the inverted index of the table and re-create the index?

**Comment by yudelevi** [comment]
@HarryLeeIBM  - Sorry, I should have added a bit more details. This was for only a portion of the data (specific tokens).
I created the index and followed up by upgrading 23.8->23.12, so this index was created before the upgrade.

I dropped/recreated the index and am not getting the error for the same tokens. I also tried querying by a token like "a" to hit many rows and didn't run into issues. I will monitor the logs and see if any more errors come up.
